programs/ziti-edge-tunnel/netif_driver/linux/resolvers.c

```diff
@@ -432,24 +457,33 @@ static void cleanup_bufferp(char **buffer) {
     }
 }
 
-void dns_update_etc_resolv(const char* tun, unsigned int ifindex, const char* addr) {
-      const char* match = "nameserver ";
-      int replace_size = strlen(match) + strlen(addr) + 1;
-      _cleanup_(cleanup_bufferp) char* replace = (char *)malloc(replace_size);
+void dns_update_etc_resolv(const char *tun, unsigned int ifindex, const char *addr) {
+      bool copy_r = make_copy(RESOLV_CONF_FILE, RESOLV_CONF_FILE ".bkp");
+
+      const char *match = "nameserver ";
+      off_t replace_size = strlen(match) + strlen(addr) + sizeof(char);
```

Comment:
This is handled in #L466. Accounting for the null byte is always handled in the allocation functions in this code so that I could keep it consistent across `malloc` and `realloc` when computing the `realloc` size in #L533. In other words, `*_size` here indicates the size _without_ the null bytes.